fperms_iscore/permissions.py

```diff
@@ -16,25 +18,56 @@
 GENERAL_CACHE_NAME = '__all__'
 
 
-def _get_perm_slug_from_data(perm_type, perm_codename, object_ct=None, object_id=None):
+def _get_perm_slug_from_data(perm_type, perm_codename, object_ct_id=None, object_id=None):
     slug = '{}|{}'.format(perm_type, perm_codename)
-    if object_ct and object_id:
-        return '{}|{}|{}'.format(slug, object_ct.pk, object_id)
+    if object_ct_id and object_id:
+        return '{}|{}|{}'.format(slug, object_ct_id, object_id)
     return slug
 
 
 def _get_perm_slug_from_obj(perm_type, perm_codename, obj=None):
     from django.contrib.contenttypes.models import ContentType
 
-    object_ct = object_id = None
+    object_ct_id = object_id = None
     if obj is not None:
-        object_ct = ContentType.objects.get_for_model(obj)
+        object_ct_id = ContentType.objects.get_for_model(obj).pk
         object_id = obj.pk
-    return _get_perm_slug_from_data(perm_type, perm_codename, object_ct, object_id)
+    return _get_perm_slug_from_data(perm_type, perm_codename, object_ct_id, object_id)
 
 
 def _get_perm_slug(perm):
-    return _get_perm_slug_from_data(perm.type, perm.codename, perm.content_type, perm.object_id)
+    return _get_perm_slug_from_data(perm.type, perm.codename, perm.content_type_id, perm.object_id)
+
+
+def get_cache():
+    return caches[settings.IS_CORE_PERM_CACHE_NAME] if settings.IS_CORE_PERM_USE_CACHE else None
+
+
+def get_all_user_perm_slugs(request):
+    if hasattr(request.user, '_fperms_is_core_user_perm_slugs'):
+        return request.user._fperms_is_core_user_perm_slugs
+
+    cache_key = None
+    cache = get_cache()
+
+    if cache:
+        assert hasattr(request, 'session'), 'The cached permissions requres session middleware to be installed, ' \
+                                            'and come before the message middleware in the MIDDLEWARE list'
+        cache_key = f'fperms_is_core-{request.user.pk}-{request.session.session_key}'
+        cache = caches[settings.IS_CORE_PERM_CACHE_NAME]
```

Comment:
don't you set it already on line 51? what is the difference? 🤔  also unlike the `get_cache` function, this doesn't check whether the setting exists

Reply:
yes this is wrong :)